File: web/src/services/sys/sysDictController.ts
// @ts-ignore
/* eslint-disable */
import { request } from '@umijs/max';

/** 此处后端没有提供注释 POST /api/sys/dict/create */
export async function createSysDict(
  body: API.SysDictCreateInput,
  options?: { [key: string]: any },
) {
  return request<API.RSysDictDetailView>('/api/sys/dict/create', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 DELETE /api/sys/dict/delete/${param0} */
export async function deleteSysDictById(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.deleteSysDictByIdParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/sys/dict/delete/${param0}`, {
    method: 'DELETE',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/dict/id/${param0} */
export async function findSysDictById(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysDictByIdParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.RSysDictDetailView>(`/api/sys/dict/id/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/dict/list */
export async function findSysDictList(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysDictListParams,
  options?: { [key: string]: any },
) {
  return request<API.RListSysDictPageView>('/api/sys/dict/list', {
    method: 'GET',
    params: {
      ...params,
      specification: undefined,
      ...params['specification'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/sys/dict/page */
export async function findSysDictPage(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.findSysDictPageParams,
  options?: { [key: string]: any },
) {
  return request<API.RListSysDictPageView>('/api/sys/dict/page', {
    method: 'GET',
    params: {
      ...params,
      specification: undefined,
      ...params['specification'],
      pageable: undefined,
      ...params['pageable'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /api/sys/dict/update */
export async function updateSysDictById(
  body: API.SysDictUpdateInput,
  options?: { [key: string]: any },
) {
  return request<API.RSysDictDetailView>('/api/sys/dict/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

File: src/main/kotlin/com/zhengchalei/xadmin/modules/sys/controller/SysDictController.kt
/*
Copyright 2024 [郑查磊]

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
package com.zhengchalei.xadmin.modules.sys.controller

import com.zhengchalei.xadmin.commons.QueryPage
import com.zhengchalei.xadmin.commons.R
import com.zhengchalei.xadmin.modules.sys.domain.Log
import com.zhengchalei.xadmin.modules.sys.domain.OperationType
import com.zhengchalei.xadmin.modules.sys.domain.dto.*
import com.zhengchalei.xadmin.modules.sys.service.SysDictService
import jakarta.validation.constraints.NotNull
import org.babyfish.jimmer.client.meta.Api
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@Api("sys")
@Validated
@RestController
@RequestMapping("/api/sys/dict")
class SysDictController(val sysDictService: SysDictService) {
    @PreAuthorize("hasAuthority('sys:dict:read') or hasAnyRole('admin')")
    @GetMapping("/id/{id}")
    fun findSysDictById(@PathVariable id: Long): R<SysDictDetailView> {
        val data = sysDictService.findSysDictById(id)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:dict:read') or hasAnyRole('admin')")
    @GetMapping("/list")
    fun findSysDictList(@NotNull specification: SysDictListSpecification): R<List<SysDictPageView>> {
        val data = sysDictService.findSysDictList(specification)
        return R(data = data)
    }

    @PreAuthorize("hasAuthority('sys:dict:read') or hasAnyRole('admin')")
    @GetMapping("/page")
    fun findSysDictPage(
        @NotNull specification: SysDictListSpecification,
        @NotNull pageable: QueryPage,
    ): R<MutableList<SysDictPageView>> {
        val data = sysDictService.findSysDictPage(specification, pageable.page())
        return R.success(data)
    }

    @Log(value = "创建系统字典", type = OperationType.CREATE)
    @PreAuthorize("hasAuthority('sys:dict:create') or hasAnyRole('admin')")
    @PostMapping("/create")
    fun createSysDict(@NotNull @RequestBody sysDictCreateInput: SysDictCreateInput): R<SysDictDetailView> {
        val data = sysDictService.createSysDict(sysDictCreateInput)
        return R(data = data)
    }

    @Log(value = "修改系统字典", type = OperationType.UPDATE)
    @PreAuthorize("hasAuthority('sys:dict:edit') or hasAnyRole('admin')")
    @PostMapping("/update")
    fun updateSysDictById(@NotNull @RequestBody sysDictUpdateInput: SysDictUpdateInput): R<SysDictDetailView> {
        val data = sysDictService.updateSysDictById(sysDictUpdateInput)
        return R(data = data)
    }

    @Log(value = "删除系统字典", type = OperationType.DELETE)
    @PreAuthorize("hasAuthority('sys:dict:delete') or hasAnyRole('admin')")
    @DeleteMapping("/delete/{id}")
    fun deleteSysDictById(@NotNull @PathVariable id: Long): R<Boolean> {
        sysDictService.deleteSysDictById(id)
        return R()
    }
}
